src/tests/ftest/util/server_utils.py

```diff
@@ -436,6 +436,10 @@ def start(self, yamlfile):
                 cmd_touch_log = "touch {}".format(lfile)
                 pcmd(self._hosts, cmd_touch_log, False)
 
+            # Change ownership of attach info file
+            chmod_attach = "chmod 777 -R {}".format(self.attach.value)
```

Comment:
Even though this is just test code, we wouldn't land this `777` mode change would we?

Also, isn't `self.attach`.value a file?  If so, there is no need for `-R` and that along with the `777` makes for a dangerous combination.

Reply:
I'm not sure, the intention here is to change the permissions or ownership of the directory so that the root user can have the ability to create a file in that directory. When I try locally, I'm not able to change the owner of the directory or create a new file using root, getting a "Operation not permitted" error. @brianjmurrell, do you think then that we can implement something more secure by modifying the permissions of the mounted shared area? As I understand it, the attach_info_path is mounted on the host, so can we specify in the mount setup for that directory to have r/w permissions for root?

Reply:
Hmm. If I understand correctly, the issue is that when daos_server starts as root in order to format NVMe, it fails to create the attach_info on the NFS share due to permissions problems. This is likely due to the share being exported with root_squash enabled, which effectively gives the daos_server process the same permissions as the "nobody" user.

One solution to this would be to chmod 777 the attach_info directory as the non-root user -- this should allow the daos_server process to create the file. It's worth noting that it may create the file with 0600 permissions, which means that it might still need to be fixed afterward in order to be read on other nodes.

All that having been said, I wonder if we're looking at the wrong problem here. I believe that the attach_info file is only used by tools (e.g. daos, dmg_old, etc) to do pool/container management after the server has already started in regular mode (i.e. after formatting is complete). In maintenance mode, the server is hanging around waiting for dmg to send it a format RPC, and that comes in via TCP.

I suggest trying this: When the server is started as root, don't supply the -a argument. This should allow the server to start in maintenance mode without attempting to write the attach_info file. When it is restarted as the non-root user, then it should be able to write to the NFS share as usual.

Reply:
I am currently handling both of those cases to change permissions of  the directory and also change permissions of the created file afterwards so it can be accessed. Also, I'm afraid your suggestion won't work in this case because currently the server is not being re-spawned as a normal user.

Reply:
Ah, ok. I missed that detail. Well, in that case the current approach is probably the best one. For the record though, you probably want 0644 rather than 0777 for the file permissions. It will still work as it is, but it makes the file world-writable as well as adding the executable bits. Might be a little confusing to future readers.

Reply:
Ok. Thanks @mjmac. I am currently still working on ironing out the solution to create NFS export on the test runner to cover the concern that @brianjmurrell brought up of having 0777 permission on the attach_info_path directory.

Reply:
@brianjmurrell: Is the 0777 permission set for the shared attach_info a major concern? It doesn't seem like a problem to me. Worst that could happen is that it gets overwritten and client tools are trying to connect to the wrong server, as far as I can tell.

Reply:
`0777` on a single file and/or dir is not a concern, particularly given the short shelf-life of the whole attach-info file concept.  My main point of concern was the `-R` argument to `chmod` and the potential for bigger badder things to happen in the event that the argument to `chmod` got munged into something other than what was intended.